apps/api/src/campaign/campaign.service.ts

```diff
@@ -678,25 +780,108 @@ export class CampaignService {
           },
         })
       }
+
+      // TODO - should be managed from the admin UI
+      const percentRaised = (actualAmount / campaign.targetAmount) * 100
+
+      if (percentRaised >= 50 && percentRaised < 90)
+        // Do not await  -> send to background
+        this.sendAmountReachedNotification(campaign, 50, actualAmount).catch((e) => console.log(e))
+      if (percentRaised >= 90 && percentRaised < 100)
+        // Do not await  -> send to background
+        this.sendAmountReachedNotification(campaign, 90, actualAmount).catch((e) => console.log(e))
+      if (percentRaised >= 100)
+        // Do not await  -> send to background
+        this.sendAmountReachedNotification(campaign, 100, actualAmount).catch((e) => console.log(e))
+    }
+  }
+
+  async sendAmountReachedNotification(
+    campaign: {
+      title: string
+      startDate: Date | null
+      endDate: Date | null
+      id: string
+      slug: string
+      targetAmount: number | null
+      state: CampaignState
+      notificationLists: NotificationList[]
+    },
+    percent: 50 | 90 | 100,
+    raisedAmount: number,
+  ) {
+    // Check if such email was sent already
+    const wasSent = await this.prisma.emailSentRegistry.findFirst({
+      where: { type: EmailType[`raised${percent}`], campaignId: campaign.id },
+    })
+
+    if (wasSent) return
+
+    const template = await this.prisma.marketingTemplates.findFirst({
+      where: {
+        name: `${percent}% Raised`,
+      },
+    })
+
+    const emailLists = campaign.notificationLists
+
+    if (template && emailLists?.length) {
+      const data: SendGridParams['SendNotificationParams'] = {
+        template_id: template.id,
+        list_ids: [emailLists[0].id],
```

Comment:
Well there always is only one notification list, but since prisma stores the relation(notificationLists) as an Array, I'm always using the first(e.g only value)